hax/hax/motr/hax.c

```diff
@@ -119,6 +119,20 @@ static void entrypoint_request_cb(struct m0_halon_interface *hi,
 				  bool first_request)
 {
 	struct hax_entrypoint_request *ep;
+	struct hax_link               *hxl;
+
+	M0_ALLOC_PTR(hxl);
+	if (hxl == NULL) {
+		M0_LOG(M0_ERROR, "Cannot allocate hax_link");
+		return;
```

Comment:
[* defect *] This may lead to crash later, so this way of handling the failure wouldn't work.

Reply:
So what do you suggest, how to handle it correctly here?

Reply:
Either preallocate (if you want to keep hax alive in OOM case) or M0_ASSERT() (the fastest way).

Reply:
We don't know the number of links beforehand to pre-allocate. So assert seems to be the only option here.

I wonder - why there is no way to return some status code from this callback function in the interface?

Reply:
Added M0_ASSERT.

Reply:
> I wonder - why there is no way to return some status code from this callback function in the interface?

To simplify the error handling. If it would be a way then the state machine of HA links would be much more complicated.